awscli/formatter.py

```diff
@@ -97,6 +97,30 @@ def _format_response(self, command_name, response, stream):
             stream.write('\n')
 
 
+class YAMLFormatter(FullyBufferedFormatter):
+    def _format_response(self, command_name, response, stream):
+        if response == {}:
+            return None
+        if not isinstance(response, (dict, set, list)):
+            # The yaml library will try to disambiguate when you dump a
+            # scalar value in a variety of ways. By default it will add
+            # an elipsis on a new line at the end of the file. While
+            # this is valid YAML and it does have its benefits,
+            # it's more useful to us to omit that. Setting indent to 0
+            # will cause it to omit the elipsis, but then it adds an
+            # extraneous newline at the beginning. So we dump to a
+            # string and trim it.
+            data = yaml.safe_dump(response, allow_unicode=True, indent=0)
```

Comment:
What is the difference between `dump` and `safe_dump`? I tried looking through their docs with not much luck.